Kernel/API/Syscall.h

```diff
@@ -135,7 +135,8 @@ enum class NeedsBigProcessLock {
     S(perf_event, NeedsBigProcessLock::Yes)                \
     S(perf_register_string, NeedsBigProcessLock::Yes)      \
     S(pipe, NeedsBigProcessLock::No)                       \
-    S(pledge, NeedsBigProcessLock::No)                     \
+    S(pledge_set_capabilities, NeedsBigProcessLock::No)    \
+    S(pledge_remove_capabilities, NeedsBigProcessLock::No) \
```

Comment:
We've tried this before I think, although I don't remember the exact reasoning why we ultimately abandoned the idea.

I myself am even questioning whether "I only need \<list\> going forward" or "I don't need \<list\> going forward" is the weirder pattern to have. It's probably more fool-proof to have the application developer keep in mind which capabilities their program is effectively keeping.

Reply:
We had a whole discussion [here](https://github.com/SerenityOS/serenity/pull/23389). But sure, doing this in userspace is more smart. We could have LibCore API to actually subtract the pledges if we want to.